pkg/controllers/applicationfailover/rb_application_failover_controller.go

```diff
@@ -174,7 +177,11 @@ func (c *RBApplicationFailoverController) evictBinding(binding *workv1alpha2.Res
 				return err
 			}
 		case policyv1alpha1.Immediately:
-			binding.Spec.RemoveCluster(cluster)
+			binding.Spec.GracefulEvictCluster(cluster, workv1alpha2.NewTaskOptions(
+				workv1alpha2.WithPurgeMode(policyv1alpha1.Immediately),
+				workv1alpha2.WithProducer(RBApplicationFailoverControllerName),
+				workv1alpha2.WithReason(workv1alpha2.EvictionReasonApplicationFailure),
+				workv1alpha2.WithGracePeriodSeconds(binding.Spec.Failover.Application.GracePeriodSeconds)))
```

Comment:
I included a grace period window of 0s for the Immediately purgeMode as otherwise if the grace period is not set (which it will not be for Immediately), the default grace period will be set to 10 minutes: https://github.com/karmada-io/karmada/blob/master/cmd/controller-manager/app/options/options.go#L226

Reply:
If we set GracePeriodSeconds to 0, the following effect will occur, as soon as the cluster is added to the eviction task, it will be cleaned up immediately, which may not be what we expected.

Therefore, I understand that it is OK to use the maximum wait time of 10 minutes here. For clusters with Immediately purgeMode, the logic for clearing the eviction queue can be the same as that for Graceful purgeMode. The difference lies in whether the old `work` is deleted immediately.

How do you think?

Reply:
Makes sense. As long as we can insure that the old work is cleared immediately before we decide to reschedule, that should be okay.